### ISSUE TITLE
php 7.2 compatibility

### ISSUE THREAD

**Issue by remicollet** [php-memcached-dev/php-memcached#358]
Yes, early report ;)

Build seems ok, but test suite fails

```
TEST 3/83 [tests/append.phpt]
========DIFF========
002+ 
002- %s: cannot append/prepend with compression turned on
========DONE========
FAIL Memcached::append() [tests/append.phpt] 
TEST 4/83 [tests/bad_construct.phpt]
========DIFF========
002+ 
003+ Notice: Undefined variable: php_errormsg in /dev/shm/BUILD/php72-php-pecl-memcached-3.0.3/NTS/tests/bad_construct.php on line 4
004+ 
002- Memcached::__construct() expects parameter 1 to be string, object given
========DONE========
FAIL Memcached construct with bad arguments [tests/bad_construct.phpt] 
TEST 36/83 [tests/incrdecr.phpt]
========DIFF========
015+ 
015- Memcached::increment(): offset cannot be a negative value
017+ 
017- Memcached::decrement(): offset cannot be a negative value
========DONE========
FAIL Memcached::increment() Memcached::decrement() [tests/incrdecr.phpt] 
TEST 38/83 [tests/incrdecr_bykey.phpt]
========DIFF========
012+ 
012- Memcached::incrementByKey(): offset cannot be a negative value
014+ 
014- Memcached::decrementByKey(): offset cannot be a negative value
========DONE========
FAIL Memcached::incrementByKey() Memcached::decrementByKey() [tests/incrdecr_bykey.phpt] 
TEST 59/83 [tests/session_badconf_servers.phpt]
========DIFF========
002+ session_start(): Failed to initialize storage module: memcached (path: )
========DONE========
FAIL Session bad configurations, invalid save path (server list) [tests/session_badconf_servers.phpt] 


Tests failed    :    5 (  6.0%) (  6.3%)
Tests passed    :   74 ( 89.2%) ( 93.7%)

```

**Comment by sodabrew** [comment]
I've opened #359 but it's blocked on travis-ci/travis-ci#7989 to confirm if the changes are complete for 7.2.

**Comment by sodabrew** [comment]
Resolved by 43bbf0080206b157b0d3f47ab1f07ede4bc80c86

**Comment by remicollet** [comment]
Thanks

Still 1 failed test:

```
TEST 59/83 [tests/session_badconf_servers.phpt]
========DIFF========
002+ session_start(): Failed to initialize storage module: memcached (path: )
========DONE========
FAIL Session bad configurations, invalid save path (server list) [tests/session_badconf_servers.phpt] 
```

Only an additional error message

**Comment by sodabrew** [comment]
Fixed the session test in 39bfab06c0c27dc620486f56a926308df5d4c819 it was a text-only change not a functional change.

**Comment by remicollet** [comment]
Thanks
